services/QuillLMS/db/big_query/views/pre_post_diagnostic_skill_group_performance_view.sql

```diff
@@ -51,9 +51,21 @@ SELECT
             AND most_recent.activity_id = activity_sessions.activity_id
             AND most_recent.completed_at = activity_sessions.completed_at
       ) AS activity_sessions ON activity_sessions.classroom_unit_id = classroom_units.id AND activity_sessions.user_id = CAST(assigned_student_id AS int64) AND activity_sessions.activity_id = activities.id AND activity_sessions.visible = true
-      LEFT OUTER JOIN special.concept_results AS concept_results          ON activity_sessions.id = concept_results.activity_session_id
-        AND concept_results.attempt_number IS NULL
-      LEFT OUTER JOIN lms.questions ON STRING(PARSE_JSON(concept_results.extra_metadata).question_uid) = questions.uid
+      LEFT OUTER JOIN (
+        SELECT
+            concept_results.activity_session_id,
+            STRING(PARSE_JSON(concept_results.extra_metadata).question_uid) AS question_uid,
+            concept_results.question_number,
+            (
+              COUNT(diagnostic_question_optimal_concepts.id) > 0 AND
+              LOGICAL_AND(CASE WHEN diagnostic_question_optimal_concepts.id IS NOT NULL THEN concept_results.correct END)
+            ) AS correct
+          FROM special.concept_results AS concept_results
+          LEFT OUTER JOIN lms.diagnostic_question_optimal_concepts ON STRING(PARSE_JSON(concept_results.extra_metadata).question_uid) = diagnostic_question_optimal_concepts.question_uid AND concept_results.concept_id = diagnostic_question_optimal_concepts.concept_id
+        WHERE concept_results.attempt_number IS NULL
+        GROUP BY concept_results.activity_session_id, concept_results.extra_metadata, concept_results.question_number
+      ) AS concept_results ON activity_sessions.id = concept_results.activity_session_id
```

Comment:
Good call.  I think I defaulted to using the same name to avoid re-working the JOIN structure, but I believe that you're right that we should prioritize clarity and minimize surprise in a query this complicated.